tests/s3/versioning/test_limits_tagging_versioning.py

```diff
@@ -221,3 +221,155 @@ def test_tag_key_min_41278(self, versioning_status):
+        else:
+            s3_cmn_lib.upload_version(self.s3_test_obj, bucket_name=self.bucket_name,
+                                      object_name=self.object_name, file_path=self.file_path,
+                                      versions_dict=self.versions)
+        latest_ver = self.versions[self.object_name]["version_history"][-1]
+        LOGGER.info("Step 2: Successfully uploaded object %s to versioned bucket %s with "
+                    "version ID %s", self.object_name, self.bucket_name, latest_ver)
+        LOGGER.info("Step 3: Perform PUT Object Tagging for %s with 10 tag key-value pairs",
+                    self.object_name)
+        resp = s3_cmn_lib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name, version_tag=self.ver_tag,
+                                             versions_dict=self.versions, tag_count=10)
+        assert_utils.assert_true(resp[0], resp)
+        put_tag = self.ver_tag[self.object_name][latest_ver][-1]
+        LOGGER.info("Step 3: Performed PUT Object Tagging for %s with 10 tag key-value pairs",
+                    self.object_name)
+        LOGGER.info("Step 4: Perform GET Object Tagging for %s with versionId=%s",
+                    self.object_name, latest_ver)
+        resp = s3_cmn_lib.get_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name, version_id=latest_ver)
+        assert_utils.assert_true(resp[0], resp)
+        get_tag = resp[1]['TagSet'][0]
+        assert_utils.assert_equal(get_tag, put_tag, "Mismatch in tag Key-Value pair."
+                                                    f"Expected: {put_tag} \n Actual: {get_tag}")
+        LOGGER.info("Step 4: Performed GET Object Tagging for %s with versionId=%s is %s",
+                    self.object_name, latest_ver, get_tag)
+        LOGGER.info("Step 5: Perform PUT Object Tagging for %s with 11 tag key-value pairs",
+                    self.object_name)
+        resp = s3_cmn_lib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
+                                             s3_ver_test_obj=self.s3_ver_obj,
+                                             bucket_name=self.bucket_name,
+                                             object_name=self.object_name, version_tag=self.ver_tag,
+                                             versions_dict=self.versions, tag_count=11)
+        assert_utils.assert_false(resp[0], resp)
+        LOGGER.info("Step 5: PUT Object Tagging for %s with 10 tag key-value pairs failed as "
```

Comment:
We can update this log message to "more than 10 tag key-value pairs failed as expected"